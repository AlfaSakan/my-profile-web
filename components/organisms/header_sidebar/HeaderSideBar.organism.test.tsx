import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HeaderSideBar from './HeaderSideBar.organism';

describe('HeaderBody component test', () => {
  it('should render correctly', () => {
    const onClickMenu = jest.fn();

    const tree = renderer
      .create(<HeaderSideBar onClickMenu={onClickMenu} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should called onClickMenu', () => {
    const onClickMenu = jest.fn();

    render(<HeaderSideBar onClickMenu={onClickMenu} />);

    fireEvent.click(screen.getByTestId('header-side-bar-menu'));

    expect(onClickMenu).toHaveBeenCalled();
  });
});
