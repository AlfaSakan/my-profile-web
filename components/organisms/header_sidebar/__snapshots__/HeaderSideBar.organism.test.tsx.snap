// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBody component test should render correctly 1`] = `
<div
  className="flex items-center bg-secondary p-3 justify-between border-r border-b"
>
  <div
    className="flex items-center"
  >
    <div
      className="rounded-full bg-gray-300 mr-4 flex items-center justify-center overflow-hidden h-10 w-10"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fill": "white",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <div
      className="flex flex-col"
    >
      <p
        className="text-base"
      />
      <p
        className="text-xs text-slate-500"
      />
    </div>
  </div>
  <div
    className="flex items-center gap-6"
  >
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fill": "#2C3333",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 9h12v2H6V9zm8 5H6v-2h8v2zm4-6H6V6h12v2z"
        />
      </svg>
    </div>
    <div
      className="cursor-pointer"
      data-testid="header-side-bar-menu"
      onClick={[MockFunction]}
    >
      <button
        className="flex items-center justify-center w-32 h-12 rounded-lg bg-primary"
        style={
          Object {
            "height": 30,
            "width": 100,
          }
        }
      >
        <p
          className="text-sm"
        >
          Buat Group
        </p>
      </button>
    </div>
  </div>
</div>
`;
